modules/likes/jetpack-likes-settings.php

```diff
@@ -0,0 +1,645 @@
+	}
+
+	/**
+	 * Returns the current state of the "WordPress.com Likes are" option.
+	 * @return boolean true if enabled sitewide, false if not
+	 */
+	public function is_enabled_sitewide() {
+		/**
+		 * Filters whether Likes are enabled by default on all posts.
+		 * true if enabled sitewide, false if not.
+		 *
+		 * @module likes
+		 *
+		 * @since 2.2.0
+		 *
+		 * @param bool $option Are Likes enabled sitewide.
+		 */
+		return (bool) apply_filters( 'wpl_is_enabled_sitewide', ! Jetpack_Options::get_option_and_ensure_autoload( 'disabled_likes', 0 ) );
+	}
+
+	public function meta_box_save( $post_id ) {
+		if ( defined('DOING_AUTOSAVE') && DOING_AUTOSAVE ) {
+			return $post_id;
+		}
+
+		if ( empty( $_POST['wpl_like_status_hidden'] ) ) {
+			return $post_id;
+		}
+
+		// Record sharing disable. Only needs to be done for WPCOM
+		if ( ! $this->in_jetpack ) {
+			if ( isset( $_POST['post_type'] ) && in_array( $_POST['post_type'], get_post_types( array( 'public' => true ) ) ) ) {
+				if ( ! isset( $_POST['wpl_enable_post_sharing'] ) ) {
+					update_post_meta( $post_id, 'sharing_disabled', 1 );
+				} else {
+					delete_post_meta( $post_id, 'sharing_disabled' );
+				}
+			}
+		}
+
+		if ( 'post' == $_POST['post_type'] ) {
+			if ( !current_user_can( 'edit_post', $post_id ) ) {
+				return $post_id;
+			}
+		}
+
+		// Record a change in like status for this post - only if it contradicts the
+		// site like setting.
+		if ( ( $this->is_enabled_sitewide() && empty( $_POST['wpl_enable_post_likes'] ) ) || ( ! $this->is_enabled_sitewide() && ! empty( $_POST['wpl_enable_post_likes'] ) ) ) {
+			update_post_meta( $post_id, 'switch_like_status', 1 );
+		} else {
+			delete_post_meta( $post_id, 'switch_like_status' );
+		}
+
+		return $post_id;
+	}
+
+	/**
+	 * Adds the 'sharing' menu to the settings menu.
+	 * Only ran if sharedaddy and publicize are not already active.
+	 */
+	function sharing_menu() {
+		add_submenu_page( 'options-general.php', esc_html__( 'Sharing Settings', 'jetpack' ), esc_html__( 'Sharing', 'jetpack' ), 'manage_options', 'sharing', array( $this, 'sharing_page' ) );
+	}
+
+	/**
+	 * Provides a sharing page with the sharing_global_options hook
+	 * so we can display the setting.
+	 * Only ran if sharedaddy and publicize are not already active.
+	 */
+	function sharing_page() {
+		$this->updated_message(); ?>
+		<div class="wrap">
+			<div class="icon32" id="icon-options-general"><br /></div>
+			<h1><?php esc_html_e( 'Sharing Settings', 'jetpack' ); ?></h1>
+			<?php
+			/** This action is documented in modules/sharedaddy/sharing.php */
+			do_action( 'pre_admin_screen_sharing' );
+			?>
+			<?php $this->sharing_block(); ?>
+		</div> <?php
+	}
+
+	/**
+	 * Returns the settings have been saved message.
+	 */
+	function updated_message() {
+		if ( isset( $_GET['update'] ) && $_GET['update'] == 'saved' )
+			echo '<div class="updated"><p>' . esc_html__( 'Settings have been saved', 'jetpack' ) . '</p></div>';
```

Comment:
Same as above, didn't introduce this code, just moved it to a separate file. Updated in 9aaef66b8485d7aa1eab7ec9eba6d4648ae72d24.